ui/src/components/CardListItem.vue

```diff
@@ -1,21 +1,168 @@
 <template>
-    <div class="card">
-        <img class="card-img-top" src="" alt="Card image cap" />
-        <div class="card-body">
-            <h5 class="card-title">Card Title</h5>
-            <p class="card-text">Sample result</p>
-            <a href="#" class="btn btn-primary">Go Somewhere</a>
+    <div v-if="locations" class="ui fluid card">
+        <div class="content">
+            <h5 class="header">Locations</h5>
+            <p class="card-text">Source:</p>
+            <canvas id="location-chart"></canvas>
+        </div>
+    </div>
+    <div v-else-if="employmentTypes" class="ui fluid card">
+        <div class="content">
+            <h5 class="header">Employment Types</h5>
+            <p class="card-text">Source:</p>
+            <canvas id="employmentType-chart"></canvas>
+        </div>
+    </div>
+    <div v-else-if="skills" class="ui fluid card">
+        <div class="content">
+            <h5 class="header">Skills</h5>
+            <p class="card-text">Source:</p>
+            <canvas id="skill-chart"></canvas>
+        </div>
+    </div>
+    <div v-else-if="isRemote" class="ui fluid card">
+        <div class="content">
+            <h5 class="header">Remote Eligible</h5>
+            <p class="card-text">Source:</p>
+            <canvas id="remote-chart"></canvas>
         </div>
     </div>
 </template>
 
 <script>
+import Chart from 'chart.js';
+
 export default {
-    name: 'CardListItem'
+    name: 'CardListItem',
+    props: ['locations', 'employmentTypes', 'skills', 'isRemote'],
+    updated() {
+        let label = [];
+        let value = [];
+        
+        if (this.locations) {
+            for (let i = 0; i < this.locations.length; i++) {
+                label.push(this.locations[i].key);
+                value.push(this.locations[i].doc_count);
+            }
```

Comment:
Instead of using for loop, try to use array methods like `map` or `filter`
```
const labels = this.locations.map(entry => entry.key);
const values = this.locations.map(entry => entry.doc_count);
```

Reply:
Used map functions instead of for loop. Will push later